### DISCUSSION TITLE
Ustawienie ceny energii

### DISCUSSION THREAD

**Discussion by qwertypl** [PiotrMachowski/Home-Assistant-custom-components-Tauron-AMIplus#190] [Q&A] [answered]
Cześć,
Ustawiając w zakładce Energia jako źródło danych encję "tauron_importer:xxxxx...x" nie mogę ustawić stałej ceny energii. Mogę to zrobić wybierając sensory jako źródło danych, ale wtedy wyświetlane są jakieś bzdury w zakładce zużytej energii. Niby prosta sprawa, taryfa G11, a jednak nie działa

**Comment by PiotrMachowski** [comment, marked as answer]
Tak, jest to w tym momencie ograniczenie w HA

**Comment by barii86** [comment]
W zasadzie mam ten sam problem. @PiotrMachowski w jaki sposób obliczasz koszt energii z danego dnia?

**Reply by PiotrMachowski** [reply]
Tak jak napisałem w odpowiedzi powyżej, nie da się aktualnie tego zrobić w panelu Energia

**Reply by barii86** [reply]
a czy nie dałoby się w Twojej integracji zrobić dodatkowego stałego pola ceny 1kWh (zmiennej) którą użytkownik będzie definiował i na jej podstawie będzie generowany czujnik/encję  (Użyj encji do śledzenia całkowitych kosztów) podstawie jej generować cenę z całego dnia?

**Reply by PiotrMachowski** [reply]
Nie jest to takie proste, ze względu na to, że dane z eLicznika pobierane są na dzień poprzedni. Jeśli ktoś jest bardzo uparty, to pewnie by był w stanie zrobić coś takiego już teraz, używając odpowiednich kwerend [sql](https://www.home-assistant.io/integrations/sql)

**Comment by elmystico** [comment]
@PiotrMachowski upewnij się proszę co do tego- wydaje mi się, że gdyby nasz dodatek zwracał wartości odczytów licznika, tudzież statystyki, z odpowiednim rodzajem danej/przypisaną jednostką [kWh], to możliwe byłoby skorzystanie z HA. Przecież wartość ceny prądu może być wartością chwilową, zapisywaną w historii.

**Reply by PiotrMachowski** [reply]
Integracja zwraca wszystko, co potrzeba

**Reply by elmystico** [reply]
![Zrzut ekranu_10-1-2025_20596](https://github.com/user-attachments/assets/72310b16-2b65-4bde-b137-efac166e6adc)

zerknij proszę na zrzut ekranu powyżej- dodałem liczniki energii jako "gaz" po to, żeby móc obliczać koszt prądu z uwzględnieniem dynamicznych zmian cen energii właśnie (G12w)
Mam taką encję z kosztem (wartość liczbowa wylicza się jeszcze z dwóch innych, mniejsza z tym, chodzi o typ)

![image](https://github.com/user-attachments/assets/98d30086-859f-44d7-a04a-fc048f3f420b)

hm właśnie patrzę dalej i 
![image](https://github.com/user-attachments/assets/74171664-14a9-4b3f-9b7c-c8a5ef07bf2b)

czyli rzeczywiście statystyki mają jednostkę taką jak powinny

Ale lipa :-/
Wiesz, generalnie wszystko można zrobić, ale chodzi o to, żeby się nie narobić wiadomo. Korzystanie z systemowej zakładki energia jest pod tym względem wygodne, szkoda

**Reply by PiotrMachowski** [reply]
Problem polega na tym, że HA nie wspiera w tym momencie obliczania kosztów dla statystyk - działa to tylko dla encji.

**Reply by elmystico** [reply]
No patrzę właśnie dalej i zaczynam nadążać mniej więcej. Zrobiłeś nawet niedawno [commit ](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Tauron-AMIplus/commit/a3d6c776946f1197f8de0d903c433ec017c33a97) :-/
Hm ciężki temat, najwygodniej byłoby podpiąć się pod statystyki `recorder` ale to mieli na danych bieżących jak się domyślam. Ktoś z HA kiedykolwiek miał jakąś wolę współpracy, żeby np spatchować ich kod gdzieś?

**Reply by PiotrMachowski** [reply]
> Zrobiłeś nawet niedawno [commit ](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Tauron-AMIplus/commit/a3d6c776946f1197f8de0d903c433ec017c33a97):-/

Ta zmiana nie robiła akurat żadnej różnicy w tej kwestii.

> Ktoś z HA kiedykolwiek miał jakąś wolę współpracy, żeby np spatchować ich kod gdzieś?

Ktoś musi poprosić o dodanie tej funkcjonalności, sami z siebie tego pewnie nie zrobią. Na szybko znalazłem takie 2 issuesy na ten temat:

https://github.com/home-assistant/core/issues/129031
https://github.com/home-assistant/core/issues/118223
